From 1e64d5aeba9e2e310ec8f248937773d3bdb275e5 Mon Sep 17 00:00:00 2001
From: Jan200101 <sentrycraft123@gmail.com>
Date: Sun, 14 Sep 2025 18:00:50 +0200
Subject: [PATCH 5/6] Set build info via options

TrenchBroom tries to get its build information via git metadata,
which is not available when building via a tarball.

Signed-off-by: Jan200101 <sentrycraft123@gmail.com>
---
 common/CMakeLists.txt              |  3 +++
 common/cmake/GenerateVersion.cmake | 10 ++++++----
 2 files changed, 9 insertions(+), 4 deletions(-)

diff --git a/common/CMakeLists.txt b/common/CMakeLists.txt
index c4e0ad248..cb30f636d 100644
--- a/common/CMakeLists.txt
+++ b/common/CMakeLists.txt
@@ -1126,6 +1126,9 @@ add_custom_target(generate_version ${CMAKE_COMMAND}
   -D SRC="${COMMON_SOURCE_DIR}/Version.h.in"
   -D DST="${CMAKE_CURRENT_BINARY_DIR}/Version.h"
   -D GIT_EXECUTABLE="${GIT_EXECUTABLE}"
+  -D BUILD_VERSION_YEAR="${BUILD_VERSION_YEAR}"
+  -D BUILD_VERSION_NUMBER="${BUILD_VERSION_NUMBER}"
+  -D BUILD_PLATFORM_NAME="${BUILD_PLATFORM_NAME}"
   -P "${CMAKE_CURRENT_SOURCE_DIR}/cmake/GenerateVersion.cmake")
 add_dependencies(common generate_version)
 
diff --git a/common/cmake/GenerateVersion.cmake b/common/cmake/GenerateVersion.cmake
index 2a904cd88..22e020047 100644
--- a/common/cmake/GenerateVersion.cmake
+++ b/common/cmake/GenerateVersion.cmake
@@ -1,9 +1,11 @@
 include("${UTILS}")
 
-get_filename_component(SRC_DIR ${SRC} DIRECTORY)
-get_git_describe("${GIT_EXECUTABLE}" "${SRC_DIR}" GIT_DESCRIBE)
-get_app_version(GIT_DESCRIBE APP_VERSION_YEAR APP_VERSION_NUMBER APP_VERSION_RC)
-get_app_version_str(APP_VERSION_YEAR APP_VERSION_NUMBER APP_VERSION_RC APP_VERSION_STR)
 get_build_platform(APP_PLATFORM_NAME)
 
+set(APP_VERSION_YEAR ${BUILD_VERSION_YEAR})
+set(APP_VERSION_NUMBER ${BUILD_VERSION_NUMBER})
+set(APP_VERSION_STR "${APP_VERSION_YEAR}.${APP_VERSION_NUMBER}")
+set(GIT_DESCRIBE "${APP_VERSION_STR}")
+set(APP_PLATFORM_NAME "${BUILD_PLATFORM_NAME}")
+
 configure_file("${SRC}" "${DST}" @ONLY)
-- 
2.51.0

